### ISSUE TITLE
🐛 Crash in PushPacket when trying to set /lsmes

### ISSUE THREAD

**Issue by CatsEyeXI** [DerpyProjectGroup/topaz#2285]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## Branch affected by issue 

`base`

## Steps to reproduce

Updated today and have had a few of these.
First one happened when I tried to do:
!exec `print("%s", player:getMainJob())` 

Second one happened when I tried to set an /lsmes

![image](https://user-images.githubusercontent.com/65316697/179137002-c91e5ae1-fe56-468f-8e19-4d1bd77f0bcc.png)

Dump:
https://drive.google.com/file/d/13sMt4lZ8duI3rS9JgF4i6CERGAwulVrl/view?usp=sharing

## Expected behavior

not crash

**Comment by WinterSolstice8** [comment]
what commit were you on? run `!build` as gm and it will tell me. thanks

**Comment by WinterSolstice8** [comment]
also, can you query your DB, and tell me if the output looks ok?

`select * from linkshells where linkshellid = 3`

I don't need to know the output, just that it looks like not garbage, mine looks something like...

```
+-------------+---------------------+-------+--------+---------------------------------------------------------------------------------------------------------------+-------------+------------+--------+
| linkshellid | name                | color | poster | message                                                                                                       | messagetime | postrights | broken |
+-------------+---------------------+-------+--------+---------------------------------------------------------------------------------------------------------------+-------------+------------+--------+
|           6 | SuperLongLinkshellw | 63351 | Test   | aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa |  1657855409 |          2 |      0 |
+-------------+---------------------+-------+--------+---------------------------------------------------------------------------------------------------------------+-------------+------------+--------+
```

**Comment by CatsEyeXI** [comment]
![image](https://user-images.githubusercontent.com/65316697/179145921-ebf1f187-2d85-4dde-b625-e7fce6b1d58e.png)
no garbage in the output. worth noting, that IS the lsmes i tried to set when it crashed...

**Comment by WinterSolstice8** [comment]
Can't reproduce the ls message crash, but the `!exec print("%s", player:getMainJob())` did crash. I do not think it should crash, but the real usage you wanted was `!exec printf("%d", player:getMainJob())`, "print" shouldn't take more than 1 param. Can you make a separate issue about the lua command crash? Lua errors like that should only print error essages and not be fatal.

The ls message crash looks like it tried to access a garbage PChar during iteration. I can look into how the LS list gets updated to see if there's any room for safeties. No promises, but I can dig around for a bit.

**Comment by CatsEyeXI** [comment]
> Can't reproduce the ls message crash, but the `!exec print("%s", player:getMainJob())` did crash. I do not think it should crash, but the real usage you wanted was `!exec printf("%d", player:getMainJob())`, "print" shouldn't take more than 1 param. Can you make a separate issue about the lua command crash? Lua errors like that should only print error essages and not be fatal.
> 
> The ls message crash looks like it tried to access a garbage PChar during iteration. I can look into how the LS list gets updated to see if there's any room for safeties. No promises, but I can dig around for a bit.

yeah i realized after sending the command that its not a string, was expecting an error not a crash though..
i think print(player:getMainJob()) would have worked also? but thats besides the point..

I'm running in debug for the night so hopefully ill catch something more useful for you to work off of. I'll open the other issue as requested, although its certainly an ID-10T error :)

**Comment by WinterSolstice8** [comment]
I think you're right that `print(player:getMainJob())` would've worked as well. and it's really easy to forget the "f" on printf so I can't blame you, and it _is_ a real problem.

**Comment by TeoTwawki** [comment]
> I think you're right that `print(player:getMainJob())` would've worked as well. and it's really easy to forget the "f" on printf so I can't blame you, and it _is_ a real problem.

would've required `toString()` or `string.format()`

we used to be able to just pass stuff straight in and print would "figure it out" as long as we didn't specifically tell it the wrong thing (which would then generate error messages about typing and nil)

just as a side note `tonumber()` is also a thing

edit: I usedtaculd speel n type a lil bit

**Comment by WinterSolstice8** [comment]
#2377 may help with this, but there is some suspicious code around deleting members in the list

```
    bool DelOnlineMember(CCharEntity* PChar, CItemLinkshell* PItemLinkshell)
    {
        XI_DEBUG_BREAK_IF(PChar == nullptr);
        if (PItemLinkshell != nullptr && PItemLinkshell->isType(ITEM_LINKSHELL))
        {
            try
            {
                CLinkshell* Linkshell = LinkshellList.at(PItemLinkshell->GetLSID()).get();
                if (!Linkshell->DelMember(PChar))
                {
                    LinkshellList.erase(PItemLinkshell->GetLSID());
                }
            }
            catch (std::out_of_range& exception)
            {
                ShowError("linkshell::DelOnlineMember caught exception: %s", exception.what());
            }
        }
        return false;
    }
```

It is very unusual to see a try/catch block in this codebase so it makes me think something strange is going on
